### PR TITLE
small improvements for readability and gas

### PR DESCRIPTION
**Pull request by zeroknots**
Found a few points for improvements during my code review. figured its easiest to take out a PR instead of making a list.

No functionality changes, but should improve on gas and readability

### CODE REVIEW

File: src/libraries/BytesLib.sol
Comment (zeroknots) [inline src/libraries/BytesLib.sol:1]: was able to axe BytesLib entirely by using calldata slicing. should reduce audit LoC

File: src/experimental/EmailAccountRecoveryNew.sol [hunk 1]
```diff
@@ -12,9 +12,15 @@ import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy
 /// new guardian and recovering a wallet.
 abstract contract EmailAccountRecoveryNew {
     uint8 constant EMAIL_ACCOUNT_RECOVERY_VERSION_ID = 1;
-    address public verifierAddr;
-    address public dkimAddr;
-    address public emailAuthImplementationAddr;
+    address internal immutable verifierAddr;
```
Comment (zeroknots) [inline src/experimental/EmailAccountRecoveryNew.sol:15]: these have getter functions. so will reduce deployment bytecode size

Comment (JohnGuilding) [reply src/experimental/EmailAccountRecoveryNew.sol:15]: makes sense, posted this suggestion in https://github.com/zkemail/ether-email-auth/pull/28

File: src/libraries/BytesLib.sol
Comment (JohnGuilding) [reply src/libraries/BytesLib.sol:1]: Ah yeah good shout since the calldata isn't stored in memory anymore. This is a good win

**Review by JohnGuilding — APPROVED** [review]
LGTM, cheers!

**Pull request merged**